### PR TITLE
doc UnifyFS as SEPARATE user file system

### PR DESCRIPTION
**Pull request by craigsteffen**
Highlight UnifyFS's separate file system nature in intro doc.

### Description

I think that the fact that UnifyFS has a completely separate 
name space and a file system that's unknown by the OS 
is something that should be spelled out more clearly.  
I think this knowledge is core to understanding how 
UnifyFS works and how people will interact with it.  

This is my first stab to write that statement.

This is a draft pull requestion.  I imagine others will want 
to talk about whether this is necessary at all or
the implementation.  This draft request will create
the space to do that.  

### Motivation and Context

The documentation is fine, but it assumes that the reader
understands the FULL implication of the words "user-side
file system", but if it's not called out in the introduction, I 
think that people will read past it and then be confused.  

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the UnifyFS code style requirements.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commit messages are properly formatted.

### CODE REVIEW

**Comment by MichaelBrim** [comment]
This needs a little wordsmithing. From your draft text, I think the main points are:
1. The UnifyFS file system is accessible only to application clients that explicitly link with our client library. Therefore, it is not possible to view the file system namespace or file contents using existing system tools or utilities, since they are not explicitly integrated.
2. The UnifyFS file system has a limited lifetime that corresponds with the lifetime of the UnifyFS servers. Typically, servers are started at the beginning of a user job and terminated at the end of the job. Files and associated data within UnifyFS cease to exist when the servers are terminated.
3. It is the user's responsibility to move files that need to be persisted to another file system. We provide an API and a utility to make this easier.

File: docs/overview.rst [hunk 1]
```diff
@@ -11,6 +11,16 @@ easily as they do the parallel file system. This section will provide a high
 level design of UnifyFS. It will describe the UnifyFS library and the UnifyFS
 daemon.
 
+The file system UnifyFS instantiates exists only in user space and is only
+visible to applications linked against the UnifyFS client library.  Since
+traditional file system tools (ls, cd, etc.) do not linked against it so they
+cannot be used to manipulate files within UnifFS.  Each UnifyFS file system
```
Comment (adammoody) [inline docs/overview.rst]: UnifyFS

File: docs/overview.rst [hunk 2]
```diff
@@ -11,6 +11,16 @@ easily as they do the parallel file system. This section will provide a high
 level design of UnifyFS. It will describe the UnifyFS library and the UnifyFS
 daemon.
 
+The file system UnifyFS instantiates exists only in user space and is only
+visible to applications linked against the UnifyFS client library.  Since
+traditional file system tools (ls, cd, etc.) do not linked against it so they
```
Comment (adammoody) [inline docs/overview.rst]: are not linked against the UnifyFS client library, they

Comment (adammoody) [reply docs/overview.rst]: The file system that UnifyFS instantiates only exists in user space

File: docs/overview.rst [hunk 3]
```diff
@@ -11,6 +11,17 @@ easily as they do the parallel file system. This section will provide a high
 level design of UnifyFS. It will describe the UnifyFS library and the UnifyFS
 daemon.
 
+The file system that UnifyFS instantiates only exists in user space and is
+only visible to applications linked against the UnifyFS client library.  Since
+traditional file system tools (ls, cd, etc.) are not linked against the
+UnifyFS client libarary they cannot see files within UnifyFS nor can they be
```
Comment (MichaelBrim) [inline docs/overview.rst]: s/libarary/library/

File: docs/overview.rst [hunk 4]
```diff
@@ -11,6 +11,17 @@ easily as they do the parallel file system. This section will provide a high
...
+used to manipulate files within UnifyFS.  Each UnifyFS file system lasts as
+long as the server processes are running, which is typically as longas the
```
Comment (MichaelBrim) [inline docs/overview.rst]: s/longas/long as/

File: docs/overview.rst [hunk 5]
```diff
@@ -11,6 +11,17 @@ easily as they do the parallel file system. This section will provide a high
...
+job they are running within.  When the servers die the file system is
```
Comment (MichaelBrim) [inline docs/overview.rst]: s/die/exit/, because we don't want to imply that our servers die a lot :)

**Comment by craigsteffen** [comment]
> The text looks good to me now. Still need to resolve those checkpatch issues (trailing space errors and commit message length warning) that Travis is failing on.

Sounds good about the text.  

Ok, I've gone into the gui on github, and from the gui editor's point of view, it very definitely doesn't have any trailing spaces.  So I'll pull it out onto a machine, collapse the commits (and fix the one with long lines), rebase it to the current repo, and then resubmit this.  And I will put in a separate PR to loosen up the requirements on trailing spaces.

**Comment by craigsteffen** [comment]
Er....it appears that I did the right thing in my branch.  I appear to have fixed the too-long commit messages, and the commit with the trailing spaces seems to have been one of the ones that was absorbed into the rebase.  

So this can be merged in, although it's several (18?) commits behind dev.  So if you want me to re-base this against the current dev before you merge, I can do that.  Or this can be merged into dev if that is easy from your point of view.

**Comment by adammoody** [comment]
Thanks @craigsteffen and @MichaelBrim !

**Pull request merged**